src/model/tls/default.rs

```diff
@@ -0,0 +1,164 @@
+        _server_name: &ServerName,
+        _scts: &mut dyn Iterator<Item = &[u8]>,
+        _ocsp_response: &[u8],
+        _now: SystemTime,
+    ) -> std::result::Result<ServerCertVerified, RustlsError> {
+        Ok(ServerCertVerified::assertion())
+    }
+}
+
+#[derive(Clone)]
+pub struct Certificate(RustlsCertificate);
+
+impl Certificate {
+    pub fn from_der(der: &[u8]) -> RdpResult<Self> {
+        Ok(Self(RustlsCertificate(der.to_vec())))
+    }
+
+    pub fn to_der(&self) -> RdpResult<Vec<u8>> {
+        Ok(self.0 .0.clone())
+    }
+}
+
+/// Copied from rustls library with removed trait bounds (Read + Write)
+/// By removing trait bounds we don't have to update
+/// other structs that depends on the Stream enum
+pub struct TlsStream<T: Sized> {
+    /// Our conneciton
+    pub conn: ClientConnection,
+
+    /// The underlying transport, like a socket
+    pub sock: T,
+}
+
+impl<T> TlsStream<T>
+where
+    T: Read + Write,
+{
+    /// Get a reference to the underlying socket
+    pub fn get_ref(&self) -> &T {
+        &self.sock
+    }
+
+    /// Get a mutable reference to the underlying socket
+    pub fn get_mut(&mut self) -> &mut T {
+        &mut self.sock
+    }
+
+    fn as_stream(&mut self) -> RustlsStream<ClientConnection, T> {
+        RustlsStream {
+            conn: &mut self.conn,
+            sock: &mut self.sock,
+        }
+    }
+}
+
+impl<T> Read for TlsStream<T>
+where
+    T: Read + Write,
+{
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        self.as_stream().read(buf)
+    }
+}
+
+impl<T> Write for TlsStream<T>
+where
+    T: Read + Write,
+{
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.as_stream().write(buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        self.as_stream().flush()
+    }
+}
+
+impl<S: fmt::Debug> fmt::Debug for TlsStream<S> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Debug::fmt(&self, fmt)
+    }
+}
+
+impl<T> TlsStream<T>
+where
+    T: Read + Write,
+{
+    pub fn new(check_certificate: bool, sock: T) -> RdpResult<Self> {
+        let root_store = RootCertStore::empty();
+        let mut config = ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(root_store)
+            .with_no_client_auth();
+
+        if !check_certificate {
+            let mut config = config.dangerous();
+            let verifier = Arc::new(DummyTlsVerifier {});
+            config.set_certificate_verifier(verifier)
+        }
+
+        config.enable_sni = false;
+        // We do not use the Server Name Indication (SNI) extension
+        // during the client handshake, but the rustls library requires
+        // a valid DNS domain name for the server regardless of that
+        // setting, so we need to provide a valid name.
+        // We can't use an empty string here.
+        let server_name: ServerName = "servername".try_into().unwrap();
```

Comment:
Maybe we can pass this through from the Go code so you don't need to put an arbitrary value there. Let's investigate this separately after the FIPS support is merged.

Reply:
We can do it, but it is not used anyway right now. In the line above you can see that we disable the SNI `config.enable_sni = false;`. So unless we do not change the SNI to true it does not matter what is in this string as long as it is not empty (this is requirement of the rustls)